client/src/main/java/io/pravega/client/stream/impl/SegmentSelector.java

```diff
@@ -82,9 +83,20 @@ public Segment getSegmentForEvent(String routingKey) {
     public List<PendingEvent> refreshSegmentEventWritersUponSealed(Segment sealedSegment, Consumer<Segment>
             segmentSealedCallback) {
         StreamSegmentsWithPredecessors successors = Futures.getAndHandleExceptions(
-                controller.getSuccessors(sealedSegment), RuntimeException::new);
-        return updateSegmentsUponSealed(currentSegments.withReplacementRange(successors), sealedSegment,
-                segmentSealedCallback);
+                controller.getSuccessors(sealedSegment), t -> {
+                    log.error("Error while fetching successors for segment: {}", sealedSegment, t);
+                    return null;
```

Comment:
This is lumping together the controller throwing an exception and the stream being deleted into the same code path. I don't think this is correct. 
It the controller throw just throw and let it be retried.

Reply:
This change is based on :
* The `controller` client has its own retry mechanism 
 `io.pravega.client.stream.impl.ControllerImpl#retryConfig` which retries on grpc exception `io.grpc.StatusRuntimeException`.

* If this exception is thrown then the `EventStreamWriter.flush()` for an existing writer instance would be blocked if the Stream is deleted, since the method `handleLogSealed` will retry on an exception.